skimage/io/tests/test_collection.py

```diff
@@ -104,29 +103,25 @@ def set_files(f):
             set_files('newfiles')
 
     @pytest.mark.skipif(not has_pooch, reason="needs pooch to download data")
-    def test_custom_load_func_sequence(self):
-        filename = fetch('data/no_time_for_that_tiny.gif')
-
+    def test_custom_load_func_sequence(self, random_gif_path):
         def reader(frameno):
-            vid = imageio.get_reader(filename)
-            return vid.get_data(frameno)
+            return imread(random_gif_path, mode="RGB")[frameno, ...]
 
         ic = ImageCollection(range(24), load_func=reader)
         # the length of ic should be that of the given load_pattern sequence
         assert len(ic) == 24
-        # GIF file has frames of size 25x14 with 4 channels (RGBA)
-        assert ic[0].shape == (25, 14, 4)
+        # GIF file has frames of size 25x14 with 3 channels (RGB)
+        assert ic[0].shape == (25, 14, 3)
 
     @pytest.mark.skipif(not has_pooch, reason="needs pooch to download data")
-    def test_custom_load_func_w_kwarg(self):
-        load_pattern = fetch('data/no_time_for_that_tiny.gif')
+    def test_custom_load_func_w_kwarg(self, random_gif_path):
 
         def load_fn(f, step):
-            vid = imageio.get_reader(f)
-            seq = [v for v in vid.iter_data()]
+            img = imread(f)
+            seq = [frame for frame in img]
```

Comment:
```suggestion
            seq = [frame for frame in iio.imiter(f)]
```

If you use `imageio.v3` (the v3 API), you will get [`imiter`](https://imageio.readthedocs.io/en/stable/_autosummary/imageio.v3.imiter.html#imageio.v3.imiter), which is exactly what you want to say here.

*Edit:* You will have to `import imageio.v3 as iio` for this. Probably wise, because you also have `skimage.io.imread`, which aliases with `imageio.v3.imread`. Adding a namespace will reduce confusion/aliasing.